packages/effect/src/internal/runtime.ts

```diff
@@ -55,6 +55,15 @@ export const unsafeFork = <R>(runtime: Runtime.Runtime<R>) =>
     runtime.runtimeFlags
   )
 
+  let effect: Effect.Effect<R, E, A> = self
+
+  if (options?.scope) {
```

Comment:
Need to add the finalizer :)

Reply:
🤦 , updated https://github.com/Effect-TS/effect/pull/1896/commits/8215ce67a8a1b0c22e4de4fdac53bbd7e98ea058